cmd/gen-go-sample/main.go

```diff
@@ -81,60 +82,61 @@ func main() {
 		donec <- struct{}{}
 	}()
 
+	// TODO(hzyi): Read sample config
+
 	if err := os.MkdirAll(*outDir, 0755); err != nil {
 		log.Fatal(err)
 	}
 
 	<-donec
 	<-donec
 
-	for _, iface := range gen.gapic.Interfaces {
-		for _, meth := range iface.Methods {
-			if err := genMethodSamples(&gen, iface, meth, *nofmt, *outDir); err != nil {
-				err = errors.E(err, "generating: %s", iface.Name+"."+meth.Name)
-				log.Fatal(err)
-			}
-		}
+	if err := genMethodSamples(&gen, gen.sampleConfig, *nofmt, *outDir); err != nil {
+		log.Fatal(err)
 	}
-}
 
-func genMethodSamples(gen *generator, iface GAPICInterface, meth GAPICMethod, nofmt bool, outDir string) error {
-	valSets := map[string]SampleValueSet{}
-	for _, vs := range meth.SampleValueSets {
-		valSets[vs.ID] = vs
-	}
+}
 
-	for _, sam := range meth.Samples.Standalone {
-		for _, vsID := range sam.ValueSets {
-			vs, ok := valSets[vsID]
-			if !ok {
-				return errors.E(nil, "value set not found: %q", vsID)
+func genMethodSamples(gen *generator, sampConf schema_v1p2.SampleConfig, nofmt bool, outDir string) error {
+	for _, samp := range sampConf.Samples {
+		for _, iface := range gen.gapic.Interfaces {
+			if iface.Name != samp.Service {
+				continue
 			}
+			for _, meth := range iface.Methods {
+				if meth.Name != samp.Rpc {
+					continue
+				}
+				gen.reset()
+				if err := gen.genSample(samp, meth); err != nil {
 
-			gen.reset()
-			if err := gen.genSample(iface.Name, meth, sam.RegionTag, vs); err != nil {
-				return errors.E(err, "value set: %s", vsID)
-			}
-			content, err := gen.commit(!nofmt, time.Now().Year())
-			if err != nil {
-				return err
-			}
+					err = errors.E(err, "generating: %s:%s", iface.Name+"."+meth.Name, samp.ID)
+					log.Fatal(err)
+				}
 
-			fname := iface.Name + "_" + meth.Name + "_" + vsID + ".go"
-			if err := ioutil.WriteFile(filepath.Join(outDir, fname), content, 0644); err != nil {
-				return err
+				content, err := gen.commit(!nofmt, time.Now().Year())
+				if err != nil {
+					return err
+				}
+				// Handle duplicate sample IDs
+				fname := samp.ID + ".go"
+				if err := ioutil.WriteFile(filepath.Join(outDir, fname), content, 0644); err != nil {
+					return err
+				}
+				break
```

Comment:
Nit: I would add a comment to remind the reader which loop we're breaking, here and below: `break //for meth`

[This pattern of having to remember a `break` at the bottom of the loop makes me slightly uneasy. If only we could do something like `defer break`..... Personally I would just up skip the non-matching elements in the loop and do the processing for the matching element outside the loop, but that's just a matter of taste]